### PR TITLE
Clean up directory structure by using WordPress Plugin Boilerplate

### PR DESCRIPTION
**Pull request by TomAbel**
For #68 
Cleaned up admin CSS from page.php to .css file.
Enqueuing styles & scripts only on the pages they are used.
Started the `@since` doc parameter at 3.0.0.

### CODE REVIEW

File: public/class-smaily-for-wp-public.php [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+ *
+ * @package    Smaily_For_WP
+ * @subpackage Smaily_For_WP/public
+ */
+
+/**
+ * The public-facing functionality of the plugin.
+ *
+ * Defines the plugin name, version, and two examples hooks for how to
+ * enqueue the public-facing stylesheet and JavaScript.
```
Comment (kaittodesk) [inline public/class-smaily-for-wp-public.php:15]: Inaccurate.

**Review by kaittodesk — CHANGES_REQUESTED** [review]
Some generic things I noticed:
- inconsistent usage of whitespace in functions and methods - sometimes function body is preceded and succeeded by a newline, sometimes only preceded, sometimes not;
- inconsistent usage of whitespace between class methods - there isn't always a newline between ending bracket and docstring start;
- not all class methods have a docstring;

**Comment by kaittodesk** [comment]
I think it is safe to remove the `includes/activator.php` file. It looks like it is not being used anywhere.

File: includes/class-smaily-for-wp.php [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+		require_once plugin_dir_path( dirname( __FILE__ ) ) . 'includes/class-smaily-for-wp-template.php';
+		/**
+		 * The class responsible for defining the widget functionality.
+		 */
+		require_once plugin_dir_path( dirname( __FILE__ ) ) . 'includes/class-smaily-for-wp-widget.php';
+		/**
+		 * The class responsible for defining all actions that occur in the public-facing
+		 * side of the site.
+		 */
+		require_once plugin_dir_path( dirname( __FILE__ ) ) . 'public/class-smaily-for-wp-public.php';
```
[the remark this answers is not in the document]
Comment (TomAbel) [reply includes/class-smaily-for-wp.php]: I noticed it as well, lots of repetitiveness in the boilerplate. c699cb6
Replaced path functions to constant where possible de40618

File: smaily-for-wp.php [hunk 3]
```diff
@@ -9,153 +9,71 @@
...
-	add_menu_page( 'smaily', 'Smaily', 'manage_options', SMLY4WP_PLUGIN_PATH, '', plugins_url( 'gfx/icon.png', __FILE__ ) );
-	add_submenu_page( 'smaily', 'Newsletter subscription form', 'Form', 'manage_options', SMLY4WP_PLUGIN_PATH, array( $template, 'dispatch' ) );
+function uninstall_smaily_for_wp() {
+	require_once plugin_dir_path( __FILE__ ) . 'includes/class-smaily-for-wp-uninstaller.php';
+	Smaily_For_WP_Uninstaller::uninstall();
 }
-add_action( 'admin_menu', 'smaily_admin_render' );
 
+register_activation_hook( __FILE__, 'activate_smaily_for_wp' );
+register_uninstall_hook( __FILE__, 'uninstall_smaily_for_wp' );
```
[the remark this answers is not in the document]
Comment (TomAbel) [reply smaily-for-wp.php]: Agree with you
7f0c6dd

File: smaily-for-wp.php [hunk 4]
```diff
@@ -9,153 +9,71 @@
...
+ * The core plugin class that is used to define internationalization,
+ * admin-specific hooks, and public-facing site hooks.
  */
-function smaily_shortcode_render( $atts ) {
-	global $wpdb;
-
-	// Load configuration data.
-	$table_name = esc_sql( $wpdb->prefix . 'smaily_config' );
-	$config = (array) $wpdb->get_row( "SELECT * FROM `$table_name` LIMIT 1" );
+require plugin_dir_path( __FILE__ ) . 'includes/class-smaily-for-wp.php';
```
[the remark this answers is not in the document]
Comment (TomAbel) [reply smaily-for-wp.php]: de40618

File: public/partials/smaily-for-wp-public-advanced.php [hunk 5]
```diff
@@ -6,17 +6,17 @@
 <?php else : ?>
 <form id="smly" action="https://<?php echo $this->domain; ?>.sendsmaily.net/api/opt-in/" method="post">
 	<p class="error" style="padding:15px;background-color:#f2dede;margin:0 0 10px;display:<?php echo $this->form_has_response ? 'block' : 'none'; ?>"><?php echo esc_html( $this->response_message ); ?></p>
-	<p class="success" style="padding:15px;background-color:#dff0d8;margin:0 0 10px;display:<?php echo $this->form_is_successful ? 'block' : 'none'; ?>"><?php echo esc_html__( 'Thank you for subscribing to our newsletter.', 'smaily-for-wp' ); ?></p>
+	<p class="success" style="padding:15px;background-color:#dff0d8;margin:0 0 10px;display:<?php echo $this->form_is_successful ? 'block' : 'none'; ?>"><?php echo esc_html__( 'Thank you for subscribing to our newsletter.', 'wp_smaily' ); ?></p>
```
[the remark this answers is not in the document]
Comment (TomAbel) [reply public/partials/smaily-for-wp-public-advanced.php:9]: Changed it back.
11c05b9

File: admin/class-smaily-for-wp-admin.php [hunk 6]
```diff
@@ -0,0 +1,417 @@
...
+
+/**
+ * The admin-specific functionality of the plugin.
+ *
+ * Defines the plugin name, version, and two examples hooks for how to
+ * enqueue the admin-specific stylesheet and JavaScript.
+ *
+ * @package    Smaily_For_WP
+ * @subpackage Smaily_For_WP/admin
+ */
```
[the remark this answers is not in the document]
Comment (TomAbel) [reply admin/class-smaily-for-wp-admin.php:19]: 8ea8005

File: includes/class-smaily-for-wp-activator.php [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+
+		// Create database table - settings.
+		$table_name            = esc_sql( $wpdb->prefix . 'smaily_config' );
+		$settings_table_exists = $wpdb->get_var(
+			$wpdb->prepare(
+				'SHOW TABLES LIKE %s',
+				$table_name
+			)
+		);
+		if ( ! $settings_table_exists ) {
```
[the remark this answers is not in the document]
Comment (TomAbel) [reply includes/class-smaily-for-wp-activator.php]: 480ca2e

File: public/class-smaily-for-wp-public.php (hunk 1 above)
Comment (TomAbel) [reply public/class-smaily-for-wp-public.php]: Removed this in f5d5fa4

File: admin/class-smaily-for-wp-admin.php [hunk 8]
```diff
@@ -0,0 +1,409 @@
...
+	 * Load subscribe widget.
+	 *
+	 * @since 3.0.0
+	 * @return void
+	 */
+	public function smaily_subscription_widget_init() {
+		register_widget( 'Smaily_For_WP_Widget' );
+	}
+
+
```
Comment (kaittodesk) [inline admin/class-smaily-for-wp-admin.php:108]: Formatting: inconsistent spacing.

File: admin/class-smaily-for-wp-admin.php [hunk 9]
```diff
@@ -0,0 +1,409 @@
...
+				$result = array(
+					'error'   => false,
+					'message' => __( 'Changes saved.', 'smaily-for-wp' ),
+				);
+				break;
+		}
+
+		// Send refresh form content (if requested).
+		if ( $refresh ) {
+
```
Comment (kaittodesk) [inline admin/class-smaily-for-wp-admin.php:387]: Formatting: inconsistent whitespace.

File: admin/class-smaily-for-wp-admin.php [hunk 10]
```diff
@@ -0,0 +1,409 @@
...
+				// Return result.
+				$result = array(
+					'error'   => false,
+					'message' => __( 'Credentials validated.', 'smaily-for-wp' ),
+				);
+				break;
+
+			case 'removeApiKey':
+
```
Comment (kaittodesk) [inline admin/class-smaily-for-wp-admin.php:257]: Formatting: inconsistent whitespace.

File: admin/class-smaily-for-wp-admin.php [hunk 11]
```diff
@@ -0,0 +1,409 @@
...
+				if ( ! empty( $insert_query ) ) {
+					$wpdb->query( "INSERT INTO `$table_name`(`id`, `title`) VALUES " . implode( ',', $insert_query ) );
+				}
+
+				// Return result.
+				$result = array(
+					'error'   => false,
+					'message' => __( 'Autoresponders refreshed.', 'smaily-for-wp' ),
+				);
+
```
Comment (kaittodesk) [inline admin/class-smaily-for-wp-admin.php:339]: Formatting: inconsistent whitespace.

File: includes/class-smaily-for-wp.php [hunk 12]
```diff
@@ -0,0 +1,225 @@
...
+	 *
+	 * Uses the Smaily_For_WP_i18n class in order to set the domain and to register the hook
+	 * with WordPress.
+	 *
+	 * @since  3.0.0
+	 * @access private
+	 */
+	private function set_locale() {
+		$plugin_i18n = new Smaily_For_WP_i18n();
+
```
Comment (kaittodesk) [inline includes/class-smaily-for-wp.php:104]: Formatting: inconsistent whitespace.

File: includes/class-smaily-for-wp.php [hunk 13]
```diff
@@ -0,0 +1,225 @@
...
+	}
+
+	/**
+	 * Install database structure (on activation).
+	 *
+	 * @since 3.0.0
+	 */
+	public static function activate() {
+		global $wpdb;
+		require_once( ABSPATH . 'wp-admin/includes/upgrade.php' );
```
Comment (kaittodesk) [inline includes/class-smaily-for-wp.php:143]: Use `require_once` consistenly - either `require_once [filename]` or `require_once( [filename] )`.

File: includes/class-smaily-for-wp.php [hunk 14]
```diff
@@ -0,0 +1,225 @@
...
+		$charset_collate = $wpdb->get_charset_collate();
+
+		// Create database table - settings.
+		$table_name            = esc_sql( $wpdb->prefix . 'smaily_config' );
+		$settings_table_exists = $wpdb->get_var(
+			$wpdb->prepare(
+				'SHOW TABLES LIKE %s',
+				$table_name
+			)
+		);
```
Comment (kaittodesk) [inline includes/class-smaily-for-wp.php:153]: This is no longer used and can be removed.

File: admin/class-smaily-for-wp-admin.php [hunk 15]
```diff
@@ -0,0 +1,409 @@
...
+		if ( isset( $screen->base ) && $screen->base === 'toplevel_page_sendsmaily-wordpress-plugin' ) {
+			wp_enqueue_style( $this->plugin_name );
+		}
+	}
+
+	/**
+	 * Register the JavaScript for the admin area.
+	 *
+	 * @since 3.0.0
+	 * @return void
```
Comment (kaittodesk) [inline admin/class-smaily-for-wp-admin.php:59]: Inconsistent usage of `@return void`.

**Review by kaittodesk — CHANGES_REQUESTED** [review]

File: admin/class-smaily-for-wp-admin.php (hunk 8 above)
Comment (TomAbel) [reply admin/class-smaily-for-wp-admin.php]: 16c786b

File: admin/class-smaily-for-wp-admin.php (hunk 9 above)
Comment (TomAbel) [reply admin/class-smaily-for-wp-admin.php]: 3b30f60

File: admin/class-smaily-for-wp-admin.php (hunk 10 above)
Comment (TomAbel) [reply admin/class-smaily-for-wp-admin.php]: fa86ce9

File: admin/class-smaily-for-wp-admin.php (hunk 11 above)
Comment (TomAbel) [reply admin/class-smaily-for-wp-admin.php]: 3b30f60

File: includes/class-smaily-for-wp.php (hunk 12 above)
Comment (TomAbel) [reply includes/class-smaily-for-wp.php]: 5417afd

File: includes/class-smaily-for-wp.php (hunk 13 above)
Comment (TomAbel) [reply includes/class-smaily-for-wp.php]: d363dde

File: includes/class-smaily-for-wp.php (hunk 14 above)
Comment (TomAbel) [reply includes/class-smaily-for-wp.php]: 96b354a

File: admin/class-smaily-for-wp-admin.php (hunk 15 above)
Comment (TomAbel) [reply admin/class-smaily-for-wp-admin.php]: removed it altogether 880eb26

**Comment by TomAbel** [comment]
Implemented the requested changes. Didn't correct or make any changes to the JavaScript as it would no longer be needed if Settings API is implemented. Plan to make a PR of it next in line.

**Review by kaittodesk — CHANGES_REQUESTED** [review]

File: admin/class-smaily-for-wp-admin.php [hunk 16]
```diff
@@ -0,0 +1,398 @@
...
+		$this->version = $version;
+	}
+
+	/**
+	 * Register the stylesheets for the admin area.
+	 *
+	 * @since 3.0.0
+	 */
+	public function enqueue_styles() {
+		wp_register_style( $this->plugin_name, SMLY4WP_PLUGIN_PATH . 'css/smaily-for-wp-admin.css', array(), $this->version, 'all' );
```
[the remark this answers is not in the document]
Comment (TomAbel) [reply admin/class-smaily-for-wp-admin.php]: Looks like this should use `SMLY4WP_PLUGIN_URL` 1402dec

File: admin/class-smaily-for-wp-admin.php [hunk 17]
```diff
@@ -0,0 +1,398 @@
...
+		}
+	}
+
+	/**
+	 * Register the JavaScript for the admin area.
+	 *
+	 * @since 3.0.0
+	 */
+	public function enqueue_scripts() {
+		wp_register_script( $this->plugin_name, SMLY4WP_PLUGIN_PATH . 'js/smaily-for-wp-admin.js', array( 'jquery' ), $this->version, false );
```
[the remark this answers is not in the document]
Comment (TomAbel) [reply admin/class-smaily-for-wp-admin.php]: Same as before, got URL and path mixed up
1402dec

File: admin/class-smaily-for-wp-admin.php [hunk 18]
```diff
@@ -0,0 +1,398 @@
...
+		$data       = $wpdb->get_row( "SELECT * FROM `$table_name` LIMIT 1" );
+		$template->assign( (array) $data );
+
+		// Load autoresponders.
+		$table_name = esc_sql( $wpdb->prefix . 'smaily_autoresponders' );
+		$data       = $wpdb->get_results( "SELECT * FROM `$table_name`" );
+		$template->assign( 'autoresponders', $data );
+
+		// Add menu elements.
+		add_menu_page( 'smaily', 'Smaily', 'manage_options', SMLY4WP_PLUGIN_PATH, '', SMLY4WP_PLUGIN_PATH . '/gfx/icon.png' );
```
[the remark this answers is not in the document]
Comment (TomAbel) [reply admin/class-smaily-for-wp-admin.php]: 1402dec

**Review by kaittodesk — APPROVED** [review]

**Pull request merged**